src/compile.c

```diff
@@ -702,13 +726,139 @@ static block bind_matcher(block matcher, block body) {
   return BLOCK(matcher, body);
 }
 
+/* Build a wrapper around a destructuring matcher so that we can chain them
+ * when we have errors.  The approach is as follows:
+ * FORK_OPT NEXT_MATCHER (unless last matcher)
+ * existing_matcher_block
+ * restructure matched variables into a new jv_object
+ * JUMP END
+ *
+ * END will destructure the object returned from a wrapper into all of the
+ * variables contained in the entire destructuring expression and is what the
+ * program's body binds to.
+ */
+static block alternation_wrapper(block submatcher, block *final_matcher, int is_first, int is_last, jv vars_list) {
+    // We need to generate an output object of what succeeded here so we can
+    // bind everyone to those values.
+    block end = gen_subexp(gen_const(jv_object()));
+
+    // Manually restructure the values we extracted into a new object.
```

Comment:
This is primarily because I couldn't find a way to `STOREV` into existing variables, though maybe that's just a lack of understanding of the bytecode format.  This is certainly still less garbage/bytecode than any existing way to implement this.

I'll take a look into making two `STOREV` instructions store to the same memory location, but I'm not sure that `inst` has a way to do that.